obspy/io/stationtxt/core.py

```diff
@@ -248,10 +247,150 @@ def _read(obj):
         inv.networks.extend(list(networks.values()))
     else:
         # Cannot really happen - just a safety measure.
-        raise NotImplementedError("Unkown level: " + str(level))
+        raise NotImplementedError("Unknown level: %s" % str(level))
     return inv
 
 
+def _format_time(value):
+    if isinstance(value, UTCDateTime):
+        return value.strftime("%Y-%m-%dT%H:%M:%S")
+
+
+def inventory_to_station_text(inventory_or_network, level):
+    """
+    Function to convert inventory or network to station text representation.
+
+    :type inventory_or_network:
+        :param:`~obspy.core.inventory.inventory.Inventory` or
+        :param:`~obspy.core.inventory.network.Network`
+        :param level: Specify level of detail using NETWORK, STATION or CHANNEL
+    """
+    if isinstance(inventory_or_network, Inventory):
+        networks = inventory_or_network.networks
+    elif isinstance(inventory_or_network, Network):
+        networks = [inventory_or_network.networks]
+    else:
+        msg = ("inventory_or_network must be a "
+               "obspy.core.inventory.network.Network or a "
+               "obspy.core.inventory.inventory.Inventory object.")
+        raise TypeError(msg)
+
+    def _to_str(item):
+        if item is None:
+            return ""
+        x = str(item)
+        if isinstance(item, UTCDateTime):
+            x = _format_time(item)
+        return x
+
+    items = []  # list of items to write
+
+    # Write items at to the requested level of detail. Raises a ValueError if
+    # insufficient information is present for the requested level of detail.
+    if level == "NETWORK":
```

Comment:
Can you add `level = level.upper()` before this switch? We're also case insensitive e.g. when specifying the write output format.. for `Stream.write(.., format='mseed'`)